test/common/upstream/health_checker_impl_test.cc

```diff
@@ -797,6 +819,30 @@ TEST_F(TcpHealthCheckerImplTest, Success) {
   read_filter_->onData(response);
 }
 
+TEST_F(TcpHealthCheckerImplTest, DataWithoutReusingConnection) {
+  InSequence s;
+
+  setupDataDontReuseConnection();
+  cluster_->hosts_ = {makeTestHost(cluster_->info_, "tcp://127.0.0.1:80")};
+  expectSessionCreate();
+  expectClientCreate();
+  EXPECT_CALL(*connection_, write(_)).Times(1);
+  EXPECT_CALL(*timeout_timer_, enableTimer(_));
+  health_checker_->start();
+
+  connection_->raiseEvent(Network::ConnectionEvent::Connected);
+
+  EXPECT_CALL(*timeout_timer_, disableTimer());
+  EXPECT_CALL(*interval_timer_, enableTimer(_));
+  EXPECT_CALL(*connection_, close(Network::ConnectionCloseType::NoFlush)).Times(1);
+
+  Buffer::OwnedImpl response;
+  add_uint8(response, 2);
+  read_filter_->onData(response);
+
+  EXPECT_EQ(1UL, cluster_->info_->stats_store_.counter("health_check.success").value());
```

Comment:
How come you don't verify these stats in TimeoutWithReuseConnection?

Reply:
Honestly I added this here because I wasn't confident I understood the flow enough for my other assertions to be right. This one was easier conceptually. I've added it to the other test as well so the next person to come along can grok things easier on first glance